### PR TITLE
`cca verify-as attester` & `cca verify-as relying-party` subcommands

### PR DESCRIPTION
**Pull request by yogeshbdeshpande**
Partially addresses #15

Signed-off-by: Yogesh Deshpande <yogesh.deshpande@arm.com>

### CODE REVIEW

File: cmd/cca/verify_as_attester.go [hunk 1]
```diff
@@ -0,0 +1,174 @@
...
+)
+
+func NewAttesterCmd(fs afero.Fs, attesterVeraisonClient common.IVeraisonClient) *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "attester",
+		Short: "Emulate an Attester",
+		Long: `This command implements the "attester mode" of a challenge-response
+interaction, where the verifier is the protocol challenger.  Therefore, the nonce or 
+challenge is provided by the Veraison API server and the CCA attestation token needs
+to be created on the fly based on the attester's claims and platform signing key (PAK)
```
Comment (thomas-fossati) [inline cmd/cca/verify_as_attester.go:47]:
```suggestion
to be created on the fly based on the attester's claims, platform signing key (PAK)
```

File: cmd/cca/verify_as_attester.go [hunk 2]
```diff
@@ -0,0 +1,174 @@
...
+			}
+
+			key, err = afero.ReadFile(fs, *realmKeyFile)
+			if err != nil {
+				return fmt.Errorf("error loading signing key from %s: %w", *realmKeyFile, err)
+			}
+
+			realmSigner, err := common.SignerFromJWK(key)
+			if err != nil {
+				return fmt.Errorf("error decoding signing key from %s: %w", *platformKeyFile, err)
```
Comment (thomas-fossati) [inline cmd/cca/verify_as_attester.go:81]:
```suggestion
				return fmt.Errorf("error decoding Realm signing key from %s: %w", *realmKeyFile, err)
```

File: cmd/cca/verify_as_attester.go [hunk 3]
```diff
@@ -0,0 +1,174 @@
...
+			}
+
+			platSigner, err := common.SignerFromJWK(key)
+			if err != nil {
+				return fmt.Errorf("error decoding signing key from %s: %w", *platformKeyFile, err)
+			}
+
+			key, err = afero.ReadFile(fs, *realmKeyFile)
+			if err != nil {
+				return fmt.Errorf("error loading signing key from %s: %w", *realmKeyFile, err)
```
Comment (thomas-fossati) [inline cmd/cca/verify_as_attester.go:76]:
```suggestion
				return fmt.Errorf("error loading Platform signing key from %s: %w", *realmKeyFile, err)
```

File: cmd/cca/verify_as_attester.go [hunk 4]
```diff
@@ -0,0 +1,174 @@
...
+
+			if err = attesterVeraisonClient.SetNonceSz(attesterNonceSz); err != nil {
+				return err
+			}
+
+			attesterVeraisonClient.SetDeleteSession(true)
+			attestationResults, err := attesterVeraisonClient.Run()
+			if err != nil {
+				return err
+			}
```
Comment (thomas-fossati) [inline cmd/cca/verify_as_attester.go:102]: we should add a bit more context around these `return err` I think

File: cmd/cca/verify_as_attester.go [hunk 5]
```diff
@@ -0,0 +1,174 @@
...
+		}
+
+		if err := eb.Rclaims.SetChallenge(nonce); err != nil {
+			return nil, "", fmt.Errorf("setting nonce: %w", err)
+		}
+
+		_, err := eb.Pclaims.GetProfile()
+		if err != nil {
+			return nil, "", fmt.Errorf("getting profile: %w", err)
+		}
```
Comment (thomas-fossati) [inline cmd/cca/verify_as_attester.go:142]: just a curiosity: why is this needed?

File: cmd/cca/verify_as_relyingparty.go [hunk 6]
```diff
@@ -0,0 +1,122 @@
...
+			if err = veraisonClient.SetEvidenceBuilder(eb); err != nil {
+				return err
+			}
+
+			veraisonClient.SetDeleteSession(true)
+
+			attestationResults, err := veraisonClient.Run()
+			if err != nil {
+				return err
+			}
```
Comment (thomas-fossati) [inline cmd/cca/verify_as_relyingparty.go:78]: same comment about context-free `return err` here as above

File: cmd/psa/verify_as_attester_test.go [hunk 7]
```diff
@@ -222,9 +223,12 @@ func Test_AttesterCmd_protocol_run_failed(t *testing.T) {
 }
 
 func Test_attesterEvidenceBuilder_BuildP2Evidence_ok(t *testing.T) {
+	signer, err := common.SignerFromJWK(testValidKey)
+	assert.NoError(t, err)
```
Comment (thomas-fossati) [inline cmd/psa/verify_as_attester_test.go:227]:
```suggestion
	require.NoError(t, err)
```

File: cmd/psa/verify_as_attester_test.go [hunk 8]
```diff
@@ -249,9 +253,12 @@ func Test_attesterEvidenceBuilder_BuildP2Evidence_ok(t *testing.T) {
 }
 
 func Test_attesterEvidenceBuilder_BuildP1Evidence_ok(t *testing.T) {
+	signer, err := common.SignerFromJWK(testValidKey)
+	assert.NoError(t, err)
```
Comment (thomas-fossati) [inline cmd/psa/verify_as_attester_test.go:257]:
```suggestion
	require.NoError(t, err)
```

File: cmd/psa/verify_as_attester_test.go [hunk 9]
```diff
@@ -276,9 +283,12 @@ func Test_attesterEvidenceBuilder_BuildP1Evidence_ok(t *testing.T) {
 }
 
 func Test_attesterEvidenceBuilder_BuildEvidence_unsupported_media_type(t *testing.T) {
+	signer, err := common.SignerFromJWK(testValidKey)
+	assert.NoError(t, err)
```
Comment (thomas-fossati) [inline cmd/psa/verify_as_attester_test.go:287]:
```suggestion
	require.NoError(t, err)
```

**Review by thomas-fossati — COMMENTED** [review]
LGTM

I've left a few inline comments.  A few of them can be quickly resolved just by accepting the inline suggestion.

File: cmd/cca/verify_as_attester.go (hunk 5 above)
Comment (yogeshbdeshpande) [reply cmd/cca/verify_as_attester.go:142]: This is just a sanity check that the Evidence Builder Claim has a valid profile else we should bail out in absence of a profile!

Comment (thomas-fossati) [reply cmd/cca/verify_as_attester.go:142]: (just to be sure I understand) wouldn't that be caught anyway later?

File: cmd/cca/verify_as_relyingparty.go (hunk 6 above)
Comment (yogeshbdeshpande) [reply cmd/cca/verify_as_relyingparty.go]: Added a bit of description to understand which method is failing now!

File: cmd/cca/verify_as_attester.go (hunk 2 above)
Comment (yogeshbdeshpande) [reply cmd/cca/verify_as_attester.go]: Resolved it and mapped correctly!

File: cmd/cca/verify_as_attester.go (hunk 3 above)
Comment (yogeshbdeshpande) [reply cmd/cca/verify_as_attester.go]: Resolved it and mapped correctly!

File: cmd/cca/verify_as_relyingparty.go (hunk 6 above)
Comment (thomas-fossati) [reply cmd/cca/verify_as_relyingparty.go]: I might be looking in the wrong place but I can't see any change in L52,57,61,... here?

Comment (yogeshbdeshpande) [reply cmd/cca/verify_as_relyingparty.go]: I think the set from the API itself is verbose so we are ok.

For Run specifically I have changed it, as Run has particularly not very good return error description

Comment (thomas-fossati) [reply cmd/cca/verify_as_relyingparty.go]: I think I didn't made my point clear.  It's not that the error strings returned from the library calls are bad/poor and therefore we need to make them better. 
apiclient has pretty good reporting, but even if it hadn't that should be fixed in apiclient rather than patched here...  The thing I'm trying to point out is that the calling sites should add context to the error reported by the library to make clearer where the error occurred in the main flow.

Comment (thomas-fossati) [reply cmd/cca/verify_as_relyingparty.go]: Rather than going in circles, I can do a proposal.

Comment (thomas-fossati) [reply cmd/cca/verify_as_relyingparty.go]: see [4db22e3](https://github.com/veraison/evcli/pull/22/commits/4db22e39ce10056683b2f3fed1eedc89cfba055a)
